core/lib/state/src/postgres/mod.rs

```diff
@@ -25,11 +25,13 @@ mod metrics;
 mod tests;
 
 /// Type alias for smart contract source code cache.
-type FactoryDepsCache = LruCache<H256, Vec<u8>>;
+type FactoryDepsCache = LruCache<H256, (Vec<u8>, MiniblockNumber)>;
```

Comment:
Bikeshedding: IMO, it would be more readable to create a struct for cache values:

```rust
struct TimestampedFactoryDep {
    bytecode: Vec<u8>,
    inserted_at: MiniblockNumber,
}
```